### PR TITLE
Reset all stores at logout

### LINKED ISSUE #545: Stores don't get cleared on logout
When switching users or servers, the stores don't get cleared out. This can lead to temporarily seeing some items from another server or user account.

The logout process should clear all the stores.

### PR DESCRIPTION
**Pull request by ThibaultNocchi**
- [x]  Add a root action to call all stores reset actions
- [x]  Add reset actions to all stores
- [x]  Factorize the default state
- [x]  Call the root reset action at logout

fix #545

### CODE REVIEW

File: store/index.ts [hunk 1]
```diff
@@ -69,3 +69,23 @@ export const mutations: MutationTree<RootState> = {
...
+export const actions: ActionTree<RootState, RootState> = {
+  async reset({ dispatch }, { clearCritical }: { clearCritical: boolean }) {
+    const promises = [];
+    promises.push(dispatch('backdrop/clearBackdrop', { root: true }));
+    promises.push(dispatch('deviceProfile/clearDeviceProfile', { root: true }));
+    promises.push(dispatch('displayPreferences/resetState', { root: true }));
+    promises.push(dispatch('homeSection/clearHomeSection', { root: true }));
+    promises.push(dispatch('page/clearPage', { root: true }));
+    promises.push(dispatch('playbackManager/stop', { root: true }));
+    if (clearCritical)
```
Comment (ferferga) [inline store/index.ts:82]: For a one line thing it's good to do it this way, but in this huge block I would use the brace style

**Review by ferferga — CHANGES_REQUESTED** [review]

File: store/index.ts (hunk 1 above)
Comment (ThibaultNocchi) [reply store/index.ts:82]: Wdyn? The if only applies to the server clear. Or I misunderstand you :p

Comment (ferferga) [reply store/index.ts:82]: I meant to scope the if block in braces:

if (whatever) {
//Whatever
}

Comment (ThibaultNocchi) [reply store/index.ts]: Done

File: store/index.ts [hunk 2]
```diff
@@ -69,3 +69,24 @@ export const mutations: MutationTree<RootState> = {
...
+    promises.push(dispatch('playbackManager/stop', { root: true }));
+    if (clearCritical) {
+      promises.push(dispatch('servers/clearServers', { root: true }));
+    }
+    promises.push(dispatch('snackbar/resetMessage', { root: true }));
+    promises.push(dispatch('tvShows/clearTvShows', { root: true }));
+    promises.push(dispatch('user/clearUser', { root: true }));
+    promises.push(dispatch('userViews/clearUserViews', { root: true }));
+
+    await Promise.all(promises);
```
Comment (ferferga) [inline store/index.ts:90]: Sorry for not pointing it out before, but why ``Promise.all``? ``await`` at each of the dispatches should also run each of them after the last one, without the need of pushes and arrays.

Comment (ThibaultNocchi) [reply store/index.ts:90]: It's not doing the same thing. Doing `await` before every dispatch means that we wait for the first one to finish before starting the second. `Promise.all` returns a promise that's resolved when all of the promises passed as argument are resolved. So here we launch the different dispatch concurrently and wait for them all to finish. The speed gain here is rather small, but there's no need to wait for each and every one of them before starting the next one ;)

**Review by ferferga — APPROVED** [review]
Tested and works!

Rebase so we can merge this :)

**Pull request merged**